airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -188,6 +190,21 @@ def inner_wrapper(self, *args, **kwargs):
 
     fallback_to_default_project_id = staticmethod(fallback_to_default_project_id)
 
+    @staticmethod
+    def catch_http_exception(func):
```

Comment:
What's the benefit of wrapping `HttpError` in `AirflowException`? I think users will understand that an `HttpError` means that a request failed 😄.

Reply:
A common practice is to make operators only return one type of error - `AirflowException`.  I did not analyze the cause. I will forward your question to my team and I will come back with an answer. This, however, will require a deeper analysis.

Reply:
It's been several months ago when we introduced it and it's mostly for debugging clarity. From what we experienced - exceptions like HttpError were wrapped in AirflowException anyway in some executors (I believe at least Sequential Executor we used for testing). And you loose the full stack trace of the original exception as it is thrown higher in the call hierarchy - retaining only the error message, but not original stack trace.

If you throw an AirflowException on the other hand - it will not be wrapped and you can actually see where it was thrown in your code including full stack trace.

I have not checked it recently - that behaviour might have changed since, we might want to test and revisit it, but in general I think it's a good idea to only throw AirflowException from execute() methods and having decorator makes it rather straightforward.

Reply:
I found one more small difference. If you throw an AirflowException exception and the task has been marked externally as success/failed, executing is considered to be successfully completed - no log entry, etc.
Reference: 
https://github.com/apache/airflow/blob/0982715a61653d21ac5ff9c1cc26e456ff713607/airflow/models/__init__.py#L1415-L1420

Reply:
👍